src/web.ts

```diff
@@ -1,22 +1,86 @@
-import { WebPlugin } from '@capacitor/core';
-import { SignInWithApplePlugin } from './definitions';
+import { WebPlugin } from "@capacitor/core";
+
+import {
+  SignInWithApplePlugin,
+  InitOptions,
+  SignInResponse,
+  SignInError,
+} from "./definitions";
+
+declare var window: any;
+
+export class SignInWithAppleWeb extends WebPlugin
+  implements SignInWithApplePlugin {
+  hasInitialized = false;
 
-export class SignInWithAppleWeb extends WebPlugin implements SignInWithApplePlugin {
   constructor() {
     super({
-      name: 'SignInWithApple',
-      platforms: ['web']
+      name: "SignInWithApple",
+      platforms: ["web"],
+    });
+  }
+
+  async Init(options: InitOptions): Promise<void> {
+    this.loadAppleScript(() => {
```

Comment:
you've declared this async, if you convert the `loadAppleScript` method to return a promise, you can just `await this.loadAppleScript()` here